core/src/loader.rs

```diff
@@ -140,7 +180,8 @@ pub fn load_token_mint(info: &AccountInfo) -> Result<(), ProgramError> {
         return Err(ProgramError::InvalidAccountData);
     }
 
-    let _mint = Mint::unpack(&info.data.borrow())?;
+    let data = info.data.borrow().to_vec();
+    let _mint = StateWithExtensionsOwned::<spl_token_2022::state::Mint>::unpack(data);
```

Comment:
assuming this works with non-token22 mints?

what happens if unpack fails? do you need to sanitize before?